coil-core/src/nonAndroidMain/kotlin/coil3/Image.kt

```diff
@@ -18,33 +19,49 @@ actual interface Image {
     actual val height: Int
     actual val shareable: Boolean
 
-    fun asBitmap(): Bitmap
-}
+    fun Canvas.onDraw()
 
+    fun asBitmap(): Bitmap {
+        val bitmap = Bitmap()
+        bitmap.allocN32Pixels(width, height)
+        Canvas(bitmap).onDraw()
+        return bitmap
+    }
+}
 @ExperimentalCoilApi
 @Data
 class BitmapImage internal constructor(
-    val bitmap: Bitmap,
+    bitmap: Bitmap,
     override val shareable: Boolean,
 ) : Image {
+    private val image =
+        try {
+            org.jetbrains.skia.Image.makeFromBitmap(bitmap)
```

Comment:
Hmm doesn't seem like we can avoid this conversion unfortunately since `Bitmap` can't draw directly to `Canvas`. Let's rename the `asBitmap` functions to `toBitmap` since they no longer wrap the underlying data and instead do a conversion.

Reply:
I removed this and I am keeping the bitmap so we can use the `BitmapPainter` and use `FilterQuality` with bitmaps